### DISCUSSION TITLE
Vercel 构建失败

### DISCUSSION THREAD

**Discussion by ghost** [HEIGE-PCloud/DoIt#134] [Q&A] [answered]
### Describe the bug 描述你遇到的错误

Build failed

### Screenshots 屏幕截图

![image](https://user-images.githubusercontent.com/61532453/125148841-b1744400-e167-11eb-937c-845991ec52f8.png)

### Build Environment 构建环境

vercel

**Comment by HEIGE-PCloud** [comment]
What version of Hugo are you using? Can you build the site correctly in local environment?

**Comment by ghost** [comment]
你会说中文吗？

**Comment by ghost** [comment]
我先是用vercel的hugo模板搭建，然后构建错误。之后我直接fork了你仓库里的plant-cloud进行构建，但都是同样的错误

**Comment by HEIGE-PCloud** [comment]
请问你使用的是什么版本的 Hugo 进行构建？在本地能否构建成功？

**Comment by ghost** [comment]
我没有在本地进行构建，我也不知道这是什么版本，我今天刚开始接触hugo。这是我的仓库你看https://github.com/fangaso/hugo

**Comment by ghost** [comment]
10:12:59.436	初始化构建环境。这可能需要几分钟才能完成
10:15:30.588	成功：完成初始化构建环境
10:15:30.588	克隆存储库...
10:15:35.870	成功：完成克隆存储库文件
10:15:37.489	安装依赖
10:15:37.491	Python 版本设置为 2.7
10:15:38.401	v12.18.0 已经安装。
10:15:38.880	现在使用节点 v12.18.0 (npm v6.14.4)
10:15:38.919	开始恢复缓存的构建插件
10:15:38.924	完成恢复缓存的构建插件
10:15:39.039	尝试使用 ruby​​ 版本 2.7.1，从环境中读取
10:15:39.995	使用 ruby​​ 版本 2.7.1
10:15:40.235	使用 PHP 5.6 版
10:15:40.263	5.2 已经安装。
10:15:40.268	使用 Swift 5.2 版
10:15:40.268	安装 Hugo 0.54.0
10:15:40.703	Hugo 静态站点生成器 v0.54.0-B1A82C61A/extended linux/amd64 BuildDate: 2019-02-01T10:04:38Z
10:15:40.704	开始恢复缓存的 go 缓存
10:15:40.706	已完成恢复缓存的 go 缓存
10:15:40.729	转到版本 go1.14.4 linux/amd64
10:15:40.733	转到版本 go1.14.4 linux/amd64
10:15:40.734	安装缺少的命令
10:15:40.735	验证运行目录
10:15:40.735	执行用户命令：hugo
10:15:40.778	错误：“/opt/buildhome/repo/themes/DoIt/layouts/taxonomy/list.html:18:1”：解析失败：模板：taxonomy/list.html:18：未关闭的操作
10:15:40.781	失败：构建命令退出，代码：255

**Comment by ghost** [comment]
10:12:59.436	Initializing build environment. This may take up to a few minutes to complete
10:15:30.588	Success: Finished initializing build environment
10:15:30.588	Cloning repository...
10:15:35.870	Success: Finished cloning repository files
10:15:37.489	Installing dependencies
10:15:37.491	Python version set to 2.7
10:15:38.401	v12.18.0 is already installed.
10:15:38.880	Now using node v12.18.0 (npm v6.14.4)
10:15:38.919	Started restoring cached build plugins
10:15:38.924	Finished restoring cached build plugins
10:15:39.039	Attempting ruby version 2.7.1, read from environment
10:15:39.995	Using ruby version 2.7.1
10:15:40.235	Using PHP version 5.6
10:15:40.263	5.2 is already installed.
10:15:40.268	Using Swift version 5.2
10:15:40.268	Installing Hugo 0.54.0
10:15:40.703	Hugo Static Site Generator v0.54.0-B1A82C61A/extended linux/amd64 BuildDate: 2019-02-01T10:04:38Z
10:15:40.704	Started restoring cached go cache
10:15:40.706	Finished restoring cached go cache
10:15:40.729	go version go1.14.4 linux/amd64
10:15:40.733	go version go1.14.4 linux/amd64
10:15:40.734	Installing missing commands
10:15:40.735	Verify run directory
10:15:40.735	Executing user command: hugo
10:15:40.778	Error: "/opt/buildhome/repo/themes/DoIt/layouts/taxonomy/list.html:18:1": parse failed: template: taxonomy/list.html:18: unclosed action
10:15:40.781	Failed: build command exited with code: 255

**Comment by HEIGE-PCloud** [comment, marked as answer]
Vercel 默认使用 `v0.58.2` 版本的 Hugo 进行构建，这个版本过于老旧，已经不再支持。

请按照 vercel 的[官方文档](https://vercel.com/guides/deploying-hugo-with-vercel)，使用 `v0.85.0` 版本的 Hugo 进行构建。

**Comment by ghost** [comment]
谢谢大佬！你会中文和英文，好厉害
